packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx

```diff
@@ -0,0 +1,249 @@
+    padding-bottom: 28px;
+    display: flex;
+`;
+const RowButtons = styled(Row)`
+    display: flex;
+    align-items: center;
+    justify-content: flex-end;
+`;
+const StyledButton = styled(Button)`
+    width: 110px;
+    margin-left: 10px;
+
+    &:first-child {
+        margin-left: 0;
+    }
+`;
+const LeftSideButton = styled(StyledButton)`
+    margin-right: auto;
+`;
+
+interface AddressToSign {
+    address: string;
+    path: string;
+    accountsSelectOpt?: Array<{ label: string; value: string }>;
+}
+type errorsMessage = 'TR_ADDRESS_IS_NOT_VALID' | 'TR_REQUIRED_FIELD';
+type Inputs = {
+    address: string;
+    path: string;
+    signMessage: string;
+    accountsSelect: { label: string; value: string; type: '' };
+    signature: string;
+};
+
+let path: AddressToSign['address'] = ''; // fast click on copy button failed if put this inside a component
+let address: AddressToSign['path'] = '';
+let accountsSelectOpt: AddressToSign['accountsSelectOpt'];
+
+const SignMessage = ({ account, isLocked }: Props) => {
+    const { sign, openModal } = useActions({
+        sign: signVerifyActions.sign,
+        openModal: modalActions.openModal,
+    });
+    if (
+        account.networkType === 'bitcoin' &&
+        typeof account.addresses?.unused[0]?.address === 'string' &&
+        typeof account.addresses?.unused[0]?.path === 'string'
+    ) {
+        const accountsSelectArr = [...account.addresses.unused, ...account.addresses.used];
+        accountsSelectOpt = accountsSelectArr.map(el => ({
+            label: el.address,
+            value: el.path,
+        }));
+    } else if (account.networkType === 'ethereum') {
+        address = account.descriptor;
+        path = account.path;
+    }
+
+    const { getValues, setValue, errors, formState, reset, control, register, watch } = useForm<
+        Inputs
+    >({
+        defaultValues: {
+            address,
+            path,
+            signMessage: '',
+            accountsSelect: { label: '', value: '', type: '' },
+            signature: '',
+        },
+        mode: 'onChange',
+    });
+    const { isValid } = formState;
+    const watchSignature = watch('signature');
+
+    useEffect(() => {
+        return () => {
+            reset(); // clear data after coin change
+            path = '';
+            address = '';
+        };
+    }, [reset]);
+
+    return (
+        <Card title={<Translation id="TR_SIGN_MESSAGE" />} isColumn>
+            <Column>
+                <Row>
+                    {account.networkType === 'bitcoin' && (
+                        <Controller
+                            as={<StyledSelect options={accountsSelectOpt} />}
+                            control={control}
+                            topLabel={<Translation id="TR_ADDRESS" />}
+                            name="accountsSelect"
+                            state={errors.accountsSelect ? 'error' : undefined}
+                            onChange={([selected]) => {
```

Comment:
i think changing option (address) should clear signature (and copy button), otherwise result will be inconsistent.
Example:
1. sign (signature is generated)
2. change address (pick different)
3. then click "Copy result" (address displayed in modal is not the one you have used to sing, the result you now have in clipboard is incorrect)

Reply:
agree